src/main/java/unicash/logic/commands/FilterCommand.java

```diff
@@ -0,0 +1,101 @@
+package unicash.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static unicash.logic.parser.CliSyntax.PREFIX_AMOUNT;
+import static unicash.logic.parser.CliSyntax.PREFIX_CATEGORY;
+import static unicash.logic.parser.CliSyntax.PREFIX_DATETIME;
+import static unicash.logic.parser.CliSyntax.PREFIX_LOCATION;
+import static unicash.logic.parser.CliSyntax.PREFIX_NAME;
+import static unicash.logic.parser.CliSyntax.PREFIX_TYPE;
+
+import unicash.commons.util.CommandUsage;
+import unicash.commons.util.ExampleGenerator;
+import unicash.commons.util.ToStringBuilder;
+import unicash.logic.UniCashMessages;
+import unicash.logic.commands.exceptions.CommandException;
+import unicash.model.Model;
+import unicash.model.transaction.predicates.TransactionContainsAllKeywordsPredicate;
+
+/**
+ * Filters the displayed Transactions according to certain parameters.
+ */
+public class FilterCommand extends Command {
+    public static final String COMMAND_WORD = "filter";
+
+    public static final String MESSAGE_USAGE = new CommandUsage.Builder()
+            .setCommandWord(COMMAND_WORD)
+            .setDescription(
+                    "Filters the transactions in UniCa$h "
+                            + "according to the specified properties. "
+                            + "Only transactions matching all specified properties. will be displayed. "
+                            + "\n\nMultiple keywords for the same property is allowed, except for Transaction "
+                            + "Type for which there can only be one keyword. A minimum of one keyword "
+                            + "must be provided."
+            )
+            .addParameter(PREFIX_NAME, "Name", true, true)
+            .addParameter(PREFIX_TYPE, "Type", true, false)
+            .addParameter(PREFIX_AMOUNT, "Amount", true, true)
+            .addParameter(PREFIX_DATETIME, "DateTime", true, true)
+            .addParameter(PREFIX_LOCATION, "Location", true, true)
+            .addParameter(PREFIX_CATEGORY, "Category", true, true)
```

Comment:
Actually this is the exact inverse of Find, and is intended to be so. If you look at the code, Find anyMatches everything and Filter only matches if everything is matched.

`find friends mcdonalds` would give me every transaction with friends as the name, category, location (e.g. _The Wine & Gourmet Friends_) and every transaction with mcdonalds as the name ... and so on.

`filter n/friends n/mcdonalds` allows the user to exactly narrow down to transactions that must contain **both** name (e.g. _Mcdonalds with friends)_. or `filter n/friends n/mcdonalds l/COM1` that must contain all parameters in order for it to match, so the user can find exactly what they are looking for, regardless of how precise. 

While adding more parameters to `find` increases its matches expansively, adding more parameters to `filter` decreases its matches conversely. I had actually originally implemented `filter` with each parameter non-variadic, but doing it this way just seems to make _sense_ somehow. It's like, if `find` can so broad then `filter` _should_ be as narrowable as possible, else it seems to be a bit asymmetric, and I think this find/filter synergy is a solid representation of the product design criteria (_...how well your features fit together to form a cohesive product_).

Either way though, the change is a simple one line expansion of the `verifyNoDuplicatePrefixesFor` method inside FilterCommandParser. But I really think that the find/filter contrast is like the _ideal_ definition of feature-cohesion, and all we would have to do is explain this in the UG/DG.

Reply:
While your `Filter` feature is indeed the inverse of `Find`, I feel like this would prove to be confusing for users, who would believe `Find` and `Filter` to be synonymous. These two features typically are represented as separate functions in typical applications (where Filter is extremely specific, down to the T, and find is a generic search, and they can be used in *conjunction*), but filters are typically represented through UI elements. To have the `Filter` and `Find` function both to exist in a CLI application as commands, would make it difficult for users to differentiate and personally I feel that you can merge the two functionalities by adding flairs/tags to the existing `Search` function rather than make a separate one for it. 

For example, `search n/friends -m` could mean to retrieve only results that *completely match*, which would already integrate a big portion of the filter function into search.